lib/pure/asyncdispatch.nim

```diff
@@ -640,45 +635,15 @@ when defined(windows) or defined(nimdoc):
 
     let ret = WSARecv(socket.SocketHandle, addr dataBuf, 1, addr bytesReceived,
                       addr flagsio, cast[POVERLAPPED](ol), nil)
-    if ret == -1:
+    # Processing of "immediate completion" operations was removed, because under
+    # heavy load it can lead to deep recursion and stack overflow.
+    # This change also dont affect speed.
+    if ret != 0:
       let err = osLastError()
       if err.int32 != ERROR_IO_PENDING:
-        if dataBuf.buf != nil:
-          dealloc dataBuf.buf
-          dataBuf.buf = nil
+        dealloc dataBuf.buf
         GC_unref(ol)
-        if flags.isDisconnectionError(err):
```

Comment:
I have made another 2 tests, one test indicates that you are right.
Both tests i have made with TCPView tool, and i dont know how it closes connection, the only thing i saw in remote side (linux machine) it was RST packet to my telnet session so it breaks connection.
So in first test i have closed connection before WSARecv started and i got an error:
WSARecv() returned **-1** and osLastError() returned **10054** (WSAECONNRESET).
In second test i have closed connection after WSARecv was completed and started waiting for data, and i got error in callback bytesCount = 0, error code = **1236** (ERROR_OPERATION_ABORTED). And this error was not handled by IsDisconnectionError() so it made an exception.